src/main/java/alfio/controller/api/admin/AdminWaitingQueueApiController.java

```diff
@@ -148,14 +151,16 @@ public ResponseEntity<Map<String, Object>> removeSubscriber(@PathVariable("event
 
     @PutMapping("/subscriber/{subscriberId}/restore")
     public ResponseEntity<Map<String, Object>> restoreSubscriber(@PathVariable("eventName") String eventName,
-                                                                     @PathVariable("subscriberId") int subscriberId,
-                                                                     Principal principal) {
+                                                                 @PathVariable("subscriberId") int subscriberId,
+                                                                 Principal principal) {
         return performStatusModification(eventName, subscriberId, principal, WaitingQueueSubscription.Status.WAITING, WaitingQueueSubscription.Status.CANCELLED);
     }
 
     private ResponseEntity<Map<String, Object>> performStatusModification(String eventName, int subscriberId,
-                                                                               Principal principal, WaitingQueueSubscription.Status newStatus,
-                                                                               WaitingQueueSubscription.Status currentStatus) {
+                                                                          Principal principal, WaitingQueueSubscription.Status newStatus,
+                                                                          WaitingQueueSubscription.Status currentStatus) {
+        accessService.checkWaitingQueueSubscriberInEvent(subscriberId, eventName);
```

Comment:
we should probably embed "checkEventOwnership" inside "checkWaitingQueueSubscriberInEvent", to avoid unnecessary calls